src/Map.cpp

```diff
@@ -144,4 +144,28 @@ void Map::printNodes() {
         std::cout << std::endl;
         nodeIter++;
     }
+}
+
+int Map::getNodeIndex(std::string regionName) {
+  auto nodeIter = nodes->begin();
+  int result = -1;
+  int index = 0;
+  while (nodeIter != nodes->end()) {
+    if (nodeIter->region == regionName) {
+      result = index;
+      break;
+    }
+    index++;
+    nodeIter++;
+  }
+}
```

Comment:
This would be much cleaner as a for loop, no need to handle two indices. I'll push a commit today to fix it.

Reply:
If possible, could you commit on a separate PR so I can get this one through?

Reply:
It's a problem for future mike, marking resolved.